src/Fix64.cs

```diff
@@ -382,66 +395,68 @@ public static Fix64 Exp(Fix64 x)
              */
 
-            // The power-series converges much faster on positive values
-            // and exp(-x) = 1/exp(x).
-            bool neg = (x.RawValue < 0);
-            if (neg) x = -x;
-
-            Fix64 result = x + One;
-            Fix64 term = x;
+            int integerPart = (int)Floor(x);
+            x = FractionalPart(x);
 
-            for (int i = 2; i < 40; i++)
+            Fix64 result = One;
+            Fix64 term = One;
+            int i = 1;
+            while (term.m_rawValue != 0)
             {
-                term = x * term / (Fix64)i;
+                term = x * term * Ln2 / (Fix64)i;
                 result += term;
-
-                if (term.RawValue == 0)
-                    break;
+                i++;
             }
 
+            result = FromRaw(result.m_rawValue << integerPart);
             if (neg) result = One / result;
 
             return result;
         }
 
-        public static Fix64 Ln(Fix64 x)
+        public static Fix64 Log2(Fix64 x)
         {
             if (x.RawValue <= 0)
                 throw new ArgumentOutOfRangeException("Non-positive value passed to Ln", "x");
 
-            int scaling = 0;
-            while (x > EPow4)
-            {
-                x /= EPow4;
-                scaling += 4;
-            }
+            // This implementation is based on Clay. S. Turner's fast binary logarithm
+            // algorithm[1].
+
+            long b = 1U << (FRACTIONAL_PLACES - 1);
+            long y = 0;
 
-            while (x < One)
+            long rawX = x.m_rawValue;
+            while (rawX < ONE)
             {
-                x *= EPow4;
-                scaling -= 4;
+                rawX <<= 1;
+                y -= ONE;
             }
 
-            Fix64 guess = new Fix64(2);
-            Fix64 delta;
-            int count = 0;
-            do
+            while (rawX >= (ONE << 1))
             {
-                // Solving e(x) = y using Newton's method
-                // f(x) = e(x) - y
-                // f'(x) = e(x)
-                Fix64 e = Exp(guess);
-                delta = (x - e) / e;
+                rawX >>= 1;
+                y += ONE;
+            }
 
-                // It's unlikely that logarithm is very large, so avoid overshooting.
-                if (delta > Three)
-                    delta = Three;
+            Fix64 z = Fix64.FromRaw(rawX);
 
-                guess += delta;
-            } while ((count++ < 10) && (delta.RawValue != 0));
+            for (int i = 0; i < FRACTIONAL_PLACES; i++)
+            {
+                z = z * z;
+                if (z.m_rawValue >= (ONE << 1))
+                {
+                    z = Fix64.FromRaw(z.m_rawValue >> 1);
+                    y += b;
+                }
+                b >>= 1;
+            }
 
-            return guess + (Fix64)scaling;
+            return Fix64.FromRaw(y);
+        }
 
+        public static Fix64 Ln(Fix64 x)
```

Comment:
All public APIs need a ///summary. I try to stay close to System.Math in descriptions. Also discussion of accuracy if it's not accurate down to Fix64.Precision (see for ex. Sin in the latest master - I just updated it yesterday).